### ISSUE TITLE
rule proposal: no-direct-mutation-state

### ISSUE THREAD

**Issue by remitbri** [yannickcr/eslint-plugin-react#201]
As suggested by @moroshko in #195 (but the issue was closed too quickly), there should be a rule that forbids direct state mutation. I.e
```js
this.state.foo = "newValue"
```
sends a warning,
whereas
```js
this.setState({foo: "newValue"})
```
or
```js
const bar = this.state.foo
```
sends none

**Comment by mkenyon** [comment]
I noticed that this is a duplicate of #133.

(Though I think it's a good idea.)
